marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.sjs

```diff
@@ -176,12 +176,25 @@ function setArtifact(artifactType, artifactName, artifact) {
 }
 
 function validateArtifact(artifactType, artifactName, artifact) {
+  if(!validateArtifactName(artifactName)){
+    let message = `Invalid ${artifactType}`;
+    if(!(artifactType == "flow" || artifactType == "stepDefinition")){
```

Comment:
This is a leaky abstraction, where the generic artifact library has knowledge of particulars about different artifact types. I think we should remove this, particularly because how much value are we getting from including the artifact type in the message? I think "Invalid name: etc..." is sufficient and it avoids this leak.